transmogrifier/sources/datacite.py

```diff
@@ -152,3 +299,17 @@ def generate_name_identifier_url(cls, name_identifier):
         else:
             base_url = ""
         return base_url + name_identifier.string
+
+    @classmethod
+    def generate_related_item_identifier_url(cls, related_item_identifier):
+        """
+        Generate a full related item identifier URL with the specified scheme.
+        Args:
+            related_item_identifier: An BeautifulSoup Tag Tag representing a Datacite
+            relatedIdentifier XML field.
+        """
+        if related_item_identifier["relatedIdentifierType"] == "DOI":
```

Comment:
Caught a bug during verification. Both of these generate identifier url functions need to use `dict.get(key)` to account for cases where there is no `IdentifierType` attribute. We should add tests for that too.

Reply:
Good catch, fixed and added tests!